### PR TITLE
Allow modifying enchant activation chance

### PR DESCRIPTION
**Pull request by Heisenburger69**
Please make sure your pull request complies with these guidelines:
- * [x] Use same formatting
- * [x] Changes must have been tested on PMMP.
- * [x] Unless it is a minor code modification, you must use an IDE.
- * [x] Have a detailed title, like "Fix CustomEnchants::getName() must be..."

#### **What does the PR change?**

This PR implements a way to modify the enchant activation chances separate from the one in the config. The default activation chance set in the config is now obtained using the getBaseChance() function in the Reactive Trait. The PR implements two additional methods, getChanceMultiplier() and setChanceMultiplier(). The default chance multiplier is 1, so there's no necessity of setting a multiplier for each enchant. 

These methods would be useful for developers to modify enchant activation chances for different players, or modify enchant activation chances based on the presence of other enchants.

I tested this PR by registering a new enchant, "Miner's Luck", which buffed the activation chances of Tool enchants present on an item, if "Miner's Luck" is present on the item.

#### **Testing Environment**

- PHP: 7.2.0 
- PMMP: 3.11.4
- OS: Windows 10

### CODE REVIEW

**Comment by DaPigGuy** [comment]
Looking closer at this PR, this would not have the intended effect of changing a player's activation chance, and instead would affect all players.

**Comment by Heisenburger69** [comment]
Resetting the chance multiplier to 1 after the enchant activates should address this issue

**Comment by DaPigGuy** [comment]
This will still cause issues

**Comment by Heisenburger69** [comment]
How so?

**Comment by DaPigGuy** [comment]
You do realize this affects every reactive enchantment for everyone?...

**Comment by Aericio** [comment]
^ didn't mean to reference

File: src/DaPigGuy/PiggyCustomEnchants/enchants/traits/ReactiveTrait.php [hunk 1]
```diff
@@ -17,6 +17,8 @@
  */
 trait ReactiveTrait
 {
+    public $chanceMultiplier = 1;
```
Comment (DaPigGuy) [inline src/DaPigGuy/PiggyCustomEnchants/enchants/traits/ReactiveTrait.php]: Missing PHPDoc/shouldn't be defined as an int

**Comment by Aericio** [comment]
fuck you git

**Pull request merged**